homeassistant/components/mqtt/mixins.py

```diff
@@ -822,6 +840,13 @@ async def _async_process_discovery_update(
             """Process discovery update."""
             try:
                 await discovery_update(payload)
+                if integration_info := payload.get(CONF_INTEGRATION):
+                    self._integration_info = MQTT_ENTITY_INTEGRATION_INFO_SCHEMA(
+                        integration_info
+                    )
+                    self.async_log_integration_info(
```

Comment:
Instead of logging this separately, and only for entities, I'd suggest to add it to the info-level logs in discovery.py